ui/hooks/useNotificationTimeouts.ts

```diff
@@ -0,0 +1,25 @@
+import { useDispatch } from 'react-redux';
+import { deleteNotificationsById } from '../store/actions';
+import { NOTIFICATIONS_EXPIRATION_DELAY } from '../helpers/constants/notifications';
+
+/**
+ * This hook is used to enforce lifecycles for snap notifications.
+ *
+ * NOTE: This hook initiates a timeout to delete a notification,
+ * but doesn't have complete control over the lifecycle of the notification.
+ * The handling of deletion of stale notifications should be handled at the component
+ * level where the notifications are rendered.
+ *
+ * @returns A function that creates a timeout to delete a notification.
+ */
+export const useSnapNotificationTimeouts = () => {
+  const dispatch = useDispatch();
+
+  const setNotificationTimeout = (id: string) => {
+    setTimeout(() => {
```

Comment:
I think the accidental click edge case is less likely than someone trying to click through their list to clear certain notifications. The experience of having to move away from the list to see it updated feels jarring. Moreover, the notifications team had mentioned they were thinking of settings for lifecycle duration, it would become even more confusing for the user then when the actual lifecycle is not in parity with its setting.